test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc

```diff
@@ -0,0 +1,130 @@
+#include "test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h"
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExtAuthz {
+namespace {
+
+Filters::Common::ExtAuthz::ResponsePtr
+makeAuthzResponse(const Filters::Common::ExtAuthz::CheckStatus status) {
+  Filters::Common::ExtAuthz::ResponsePtr response =
+      std::make_unique<Filters::Common::ExtAuthz::Response>();
+  response->status = status;
+  // TODO: We only add the response status.
+  // Add fuzzed inputs for headers_to_(set/append/add), body, status_code to the Response.
+  return response;
+}
+
+Filters::Common::ExtAuthz::CheckStatus resultCaseToCheckStatus(
+    const envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase::AuthResult result) {
+  Filters::Common::ExtAuthz::CheckStatus check_status;
+  switch (result) {
+  case envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase::OK: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::OK;
+    break;
+  }
+  case envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase::ERROR: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::Error;
+    break;
+  }
+  case envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase::DENIED: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::Denied;
+    break;
+  }
+  default: {
+    // Unhandled status.
+    NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
+  }
+  }
+  return check_status;
+}
+
+class FuzzerMocks {
+public:
+  FuzzerMocks() : addr_(std::make_shared<Network::Address::PipeInstance>("/test/test.sock")) {
+
+    ON_CALL(decoder_callbacks_, connection()).WillByDefault(Return(&connection_));
+    connection_.stream_info_.downstream_connection_info_provider_->setRemoteAddress(addr_);
+    connection_.stream_info_.downstream_connection_info_provider_->setLocalAddress(addr_);
+  }
+
+  NiceMock<Runtime::MockLoader> runtime_;
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks_;
+  NiceMock<Http::MockStreamEncoderFilterCallbacks> encoder_callbacks_;
+  Network::Address::InstanceConstSharedPtr addr_;
+  NiceMock<Envoy::Network::MockConnection> connection_;
+};
+
+DEFINE_PROTO_FUZZER(const envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase& input) {
+  try {
+    TestUtility::validate(input);
+  } catch (const EnvoyException& e) {
+    ENVOY_LOG_MISC(debug, "EnvoyException during validation: {}", e.what());
+    return;
+  }
+
+  static FuzzerMocks mocks;
+  NiceMock<Stats::MockIsolatedStatsStore> stats_store;
+  envoy::config::bootstrap::v3::Bootstrap bootstrap;
+  Http::ContextImpl http_context(stats_store.symbolTable());
+  Filters::Common::ExtAuthz::MockClient* client = new Filters::Common::ExtAuthz::MockClient();
```

Comment:
The instantiation of `client` and `filter` below should be performed after the try-catch to avoid ASAN issues.